datastream-server/src/main/java/com/linkedin/datastream/server/zk/KeyBuilder.java

```diff
@@ -373,4 +382,22 @@ public static String getTargetAssignmentPath(String cluster, String connectorTyp
   public static String getTargetAssignmentBase(String cluster, String connectorType) {
     return String.format(TARGET_ASSIGNMENT_BASE, cluster, connectorType).replaceAll("//", "/'");
   }
+
+  /**
+   * Get all the throughput violating datastreams
+   * @param cluster Brooklin cluster name
+   * @return throughput violating datastreams
+   */
+  public static String throughputViolations(String cluster) {
+    return String.format(THROUGHPUT_VIOLATIONS, cluster);
+  }
+
+  /**
+   * Get all the throughput violating topics per datastream
+   * @param cluster Brooklin cluster name
+   * @return throughput violating topics per datastream
+   */
+  public static String throughputViolationsPerDatastream(String cluster, String datastream) {
+    return String.format(THROUGHPUT_VIOLATIONS_PER_DATASTREAM, cluster, datastream);
```

Comment:
Is this right? We discussed not adding another node at the cluster level.

We will keep it either inside the datastream metadata or another node inside the datastream.